x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java

```diff
@@ -85,28 +92,76 @@ protected void masterOperation(
             return;
         }
 
-        try {
-            ListSearchApplicationAction.Response resp = clientWithOrigin.execute(
-                ListSearchApplicationAction.INSTANCE,
-                new ListSearchApplicationAction.Request(null, new PageParams(0, 0))
-            ).get();
+        Map<String, Object> searchApplicationsUsage = new HashMap<>();
+
+        ActionListener<ListSearchApplicationAction.Response> searchApplicationsCountListener = ActionListener.wrap(response -> {
+            addSearchApplicationsUsage(response, searchApplicationsUsage);
             listener.onResponse(
                 new XPackUsageFeatureResponse(
                     new EnterpriseSearchFeatureSetUsage(
                         enabled,
                         LicenseUtils.LICENSED_ENT_SEARCH_FEATURE.checkWithoutTracking(licenseState),
-                        Map.of("count", resp.queryPage().count())
+                        searchApplicationsUsage
                     )
                 )
             );
-        } catch (Exception e) {
+        }, e -> {
             logger.warn("Failed to get search application count to include in Enterprise Search usage", e);
-            EnterpriseSearchFeatureSetUsage usage = new EnterpriseSearchFeatureSetUsage(
-                LicenseUtils.LICENSED_ENT_SEARCH_FEATURE.checkWithoutTracking(licenseState),
-                enabled,
-                Collections.emptyMap()
+            listener.onResponse(
+                new XPackUsageFeatureResponse(
+                    new EnterpriseSearchFeatureSetUsage(
+                        enabled,
+                        LicenseUtils.LICENSED_ENT_SEARCH_FEATURE.checkWithoutTracking(licenseState),
+                        searchApplicationsUsage
+                    )
+                )
             );
-            listener.onResponse(new XPackUsageFeatureResponse(usage));
+        });
+
+        ListSearchApplicationAction.Request searchApplicationsCountRequest = new ListSearchApplicationAction.Request(
+            null,
+            new PageParams(0, 10_000)
```

Comment:
Why are we setting `PageParams` to 10,000 here? Couldn't we just use the total results count instead?

Reply:
i'm actually using the results now so i can get the search application name and then use the indices to fetch field capabilities. i used `10_000` because that's what i saw [used in the ml usage transport action](https://github.com/search?q=repo%3Aelastic%2Felasticsearch+10_000+path%3A%2F%5Ex-pack%5C%2Fplugin%5C%2Fml%5C%2F%2F+path%3A%2F%5Ex-pack%5C%2Fplugin%5C%2Fml%5C%2Fsrc%5C%2Fmain%5C%2Fjava%5C%2Forg%5C%2Felasticsearch%5C%2Fxpack%5C%2Fml%5C%2F%2F+NOT+path%3A%2F%5Ex-pack%5C%2Fplugin%5C%2Fml%5C%2Fsrc%5C%2Fmain%5C%2Fjava%5C%2Forg%5C%2Felasticsearch%5C%2Fxpack%5C%2Fml%5C%2Finference%5C%2F%2F+NOT+path%3A%2F%5Ex-pack%5C%2Fplugin%5C%2Fml%5C%2Fsrc%5C%2Fmain%5C%2Fjava%5C%2Forg%5C%2Felasticsearch%5C%2Fxpack%5C%2Fml%5C%2Futils%5C%2Fpersistence%5C%2F%2F+NOT+path%3A%2F%5Ex-pack%5C%2Fplugin%5C%2Fml%5C%2Fsrc%5C%2Fmain%5C%2Fjava%5C%2Forg%5C%2Felasticsearch%5C%2Fxpack%5C%2Fml%5C%2Fannotations%5C%2F%2F+NOT+path%3A%2F%5Ex-pack%5C%2Fplugin%5C%2Fml%5C%2Fsrc%5C%2Fmain%5C%2Fjava%5C%2Forg%5C%2Felasticsearch%5C%2Fxpack%5C%2Fml%5C%2Fjob%5C%2Fprocess%5C%2F%2F+NOT+path%3A%2F%5Ex-pack%5C%2Fplugin%5C%2Fml%5C%2Fsrc%5C%2Fmain%5C%2Fjava%5C%2Forg%5C%2Felasticsearch%5C%2Fxpack%5C%2Fml%5C%2Faction%5C%2F%2F+NOT+path%3A%2F%5Ex-pack%5C%2Fplugin%5C%2Fml%5C%2Fsrc%5C%2Fmain%5C%2Fjava%5C%2Forg%5C%2Felasticsearch%5C%2Fxpack%5C%2Fml%5C%2Faggs%5C%2Fchangepoint%5C%2F%2F&type=code)

Reply:
I don't think we need that, since we return the total count

Reply:
but wouldn't this always be empty then? am i missing something?

https://github.com/elastic/elasticsearch/pull/95702/files#diff-b0631ca31ff74c153c40124ece9c0e37cb067bdae501d6561051e846faede609R133

Reply:
The results would be empty, but the [count](https://github.com/elastic/elasticsearch/blob/61c446dca23b34962b2f3e64c3b9af3d90bcec5d/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/action/util/QueryPage.java#LL36C11-L36C11) is also available.

Reply:
right i need the results because i iterate through them here

https://github.com/sloanelybutsurely/elasticsearch/blob/d93d24b21b4040160c0292fcfe5c057c84c8d5f8/x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java#L131-L142